core/src/main/scala/swave/core/StreamOps.scala

```diff
@@ -312,6 +312,46 @@ trait StreamOps[A] extends Any { self ⇒
   final def slice(startIndex: Long, length: Long): Repr[A] =
     via(Pipe[A] drop startIndex take length named "slice")
 
+  final def sliding(windowSize: Int, emitSingleEmpty: Boolean = false): Repr[immutable.Seq[A]] =
+    slidingTo[immutable.Seq](windowSize, emitSingleEmpty)
+
+  final def slidingTo[M[+ _]](windowSize: Int, emitSingleEmpty: Boolean = false)(
+      implicit cbf: CanBuildFrom[M[A], A, M[A]]): Repr[M[A]] = {
+
+    val builder = cbf.apply()
+
+    prefixAndTail(windowSize).map {
+      case (prefix, tail) =>
+        if (prefix.length == windowSize) {
+          // there are enough elements to fill at least one window
+          // prepare buffer
+          val buffer = new RingBuffer[A](roundUpToPowerOf2(windowSize))
+          prefix.foreach(x => buffer.write(x))
+
+          tail
+            .scan(buffer) { (buf, elem) ⇒
+              if (buf.size == windowSize) buf.unsafeDropHead()
+              buf.unsafeWrite(elem)
+              buf
+            }
+            .map { b =>
+              builder.clear()
+              b.foreach { builder += _ }
+              builder.result()
+            }
+        } else if (prefix.length > 0) {
+          // less elements than one window - just return the elements
+          prefix.foreach { builder += _ }
+          Spout.one(builder.result())
+        } else {
+          if (emitSingleEmpty)
```

Comment:
I think we should be able to simply drop the `emitSingleEmpty` functionality altogether.
It only affects the very end of the transformation and can be achieved like this (for example) if it is really needed:

```scala
sliding(4).nonEmptyOr(Spout.one(Nil))
```

WDYT?